adapters/adapterstest/test_json.go

```diff
@@ -53,30 +55,44 @@ import (
 //	  adapterstest.RunJSONBidderTest(t, "{bidder}test", instanceOfYourBidder)
 //	}
 func RunJSONBidderTest(t *testing.T, rootDir string, bidder adapters.Bidder) {
-	runTests(t, fmt.Sprintf("%s/exemplary", rootDir), bidder, false, false, false)
-	runTests(t, fmt.Sprintf("%s/supplemental", rootDir), bidder, true, false, false)
-	runTests(t, fmt.Sprintf("%s/amp", rootDir), bidder, true, true, false)
-	runTests(t, fmt.Sprintf("%s/video", rootDir), bidder, false, false, true)
-	runTests(t, fmt.Sprintf("%s/videosupplemental", rootDir), bidder, true, false, true)
-}
+	err := filepath.WalkDir(rootDir, func(path string, info fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
 
-// runTests runs all the *.json files in a directory. If allowErrors is false, and one of the test files
-// expects errors from the bidder, then the test will fail.
-func runTests(t *testing.T, directory string, bidder adapters.Bidder, allowErrors, isAmpTest, isVideoTest bool) {
-	if specFiles, err := os.ReadDir(directory); err == nil {
-		for _, specFile := range specFiles {
-			fileName := fmt.Sprintf("%s/%s", directory, specFile.Name())
-			specData, err := loadFile(fileName)
+		var approvedPath, allowErrors, isAmpTest, isVideoTest bool
+		switch filepath.Base(filepath.Dir(path)) {
+		case "amp":
+			isAmpTest = true
+			fallthrough
+		case "supplemental":
+			allowErrors = true
+			approvedPath = true
+
```

Comment:
Only supplemental and exemplary lack a `fallthrough`. Why is this?

Reply:
Oh, I think I see the reason now, you are execuding code from multiple cases in one pass.

Reply:
> Oh, I think I see the reason now, you are execuding code from multiple cases in one pass.

Yes but it kind of looks awful. I'm thinking about rolling back to:
```
	switch filepath.Base(filepath.Dir(path)) {
	case "exemplary":
		validDir = true
	case "supplemental":
		validDir = true
		allowErrors = true
	case "amp":
		validDir = true
		allowErrors = true
		isAmpTest = true
	case "video":
		validDir = true
		isVideoTest = true
	case "videosupplemental":
		validDir = true
		allowErrors = true
		isVideoTest = true
	}
```
What do you think

Reply:
Agreed that looks awful. Is there to factor out the importance of the directory name? Isn't that just for organization?

Reply:
It's a bit more than only organizational. The test framework in its current state fails a JSON test if said file comes with a `"expectedMakeRequestsErrors"` or a `"expectedMakeBidsErrors"` field but the `allowErrors` flag is set to `false`. Tests in the `supplemental/`, `amp/`, and `videosupplemental/` directories allow errors, but the ones in the `exemplary/` directory, not. I believe in the future we could revisit this.

Reply:
Rolled back to the more verbose but less confusing version of the `switch` statement